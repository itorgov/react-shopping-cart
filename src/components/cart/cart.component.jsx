import React from 'react';

import {CartOrder} from "../cart-order/cart-order.component";
import {CartSummaryBlock} from "../cart-summary-block/cart-summary-block.component";

import styles from './cart.module.scss';

export const Cart = () => (
    <div className={styles.cart}>
        <CartOrder/>
        <CartSummaryBlock/>
    </div>
);
