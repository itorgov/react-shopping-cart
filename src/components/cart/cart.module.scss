.cart {
  width: 100%;
  max-width: 114rem;
  margin: 5rem auto;
  background-color: var(--color-white);
  border-radius: 2px;
  box-shadow: 2rem 2rem 5rem rgba(var(--color-secondary-rgb), 0.3);

  display: flex;

  @media only screen and (max-width: 43.75em) {
    margin: 0;
    min-height: 100vh;
    flex-direction: column;
  }
}
