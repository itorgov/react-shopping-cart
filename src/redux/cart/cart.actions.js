import CartActionTypes from './cart.types';

export const addItem = item => ({
    type: CartActionTypes.ADD_ITEM,
    payload: item
});

export const incrementItemQuantity = item => ({
    type: CartActionTypes.INCREMENT_ITEM_QUANTITY,
    payload: item
});

export const decrementItemQuantity = item => ({
    type: CartActionTypes.DECREMENT_ITEM_QUANTITY,
    payload: item
});

export const removeItem = item => ({
    type: CartActionTypes.REMOVE_ITEM,
    payload: item
});

export const updateNote = text => ({
    type: CartActionTypes.UPDATE_NOTE,
    payload: text
});
