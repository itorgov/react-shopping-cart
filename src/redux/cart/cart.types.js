const CartActionTypes = {
    ADD_ITEM: 'ADD_ITEM',
    INCREMENT_ITEM_QUANTITY: 'INCREMENT_ITEM_QUANTITY',
    DECREMENT_ITEM_QUANTITY: 'DECREMENT_ITEM_QUANTITY',
    REMOVE_ITEM: 'REMOVE_ITEM',
    UPDATE_NOTE: 'UPDATE_NOTE',
};

export default CartActionTypes;
