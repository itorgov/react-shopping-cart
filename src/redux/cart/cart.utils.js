export const addItem = (items, itemToAdd) => {
    const existingCartItem = items.find(
        cartItem => cartItem.id === itemToAdd.id
    );

    if (existingCartItem) {
        return items.map(
            cartItem => cartItem.id === itemToAdd.id
                ? {...cartItem, quantity: cartItem.quantity + 1}
                : cartItem
        );
    }

    return [...items, {...itemToAdd, quantity: 1}];
};

export const incrementItemQuantity = (items, item) => {
    return items.map(
        cartItem => cartItem.id === item.id
            ? {...cartItem, quantity: cartItem.quantity + 1}
            : cartItem
    );
};

export const decrementItemQuantity = (items, item) => {
    return items.map(
        cartItem => cartItem.id === item.id && cartItem.quantity > 1
            ? {...cartItem, quantity: cartItem.quantity - 1}
            : cartItem
    );
};

export const removeItem = (items, itemToRemove) => {
    return items.filter(cartItem => cartItem.id !== itemToRemove.id);
};
