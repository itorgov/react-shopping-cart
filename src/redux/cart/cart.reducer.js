import CartActionTypes from './cart.types';
import {addItem, decrementItemQuantity, incrementItemQuantity, removeItem} from './cart.utils';

const INITIAL_STATE = {
    items: [
        {
            id: 3,
            name: "Translucent Loose Setting Powder",
            price: 3900,
            quantity: 1,
        },
        {
            id: 4,
            name: "Mini Summer Makeup Essentials Set",
            price: 8800,
            quantity: 2,
        }
    ],
    note: '',
};

const cartReducer = (state = INITIAL_STATE, action) => {
    switch (action.type) {
        case CartActionTypes.ADD_ITEM:
            return {
                ...state,
                items: addItem(state.items, action.payload)
            };
        case CartActionTypes.INCREMENT_ITEM_QUANTITY:
            return {
                ...state,
                items: incrementItemQuantity(state.items, action.payload)
            };
        case CartActionTypes.DECREMENT_ITEM_QUANTITY:
            return {
                ...state,
                items: decrementItemQuantity(state.items, action.payload)
            };
        case CartActionTypes.REMOVE_ITEM:
            return {
                ...state,
                items: removeItem(state.items, action.payload)
            };
        case CartActionTypes.UPDATE_NOTE:
            return {
                ...state,
                note: action.payload,
            };
        default:
            return state;
    }
};

export default cartReducer;
