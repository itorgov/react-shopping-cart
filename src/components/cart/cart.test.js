import React from "react";
import {shallow} from "enzyme";

import {Cart} from './cart.component';

describe('<Cart />', () => {
    it('expect to render', () => {
        expect(shallow(<Cart />)).toMatchSnapshot();
    });
});
