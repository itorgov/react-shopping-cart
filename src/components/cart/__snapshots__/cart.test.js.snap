// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> expect to render 1`] = `
<div
  className="cart"
>
  <CartOrder />
  <CartSummaryBlock />
</div>
`;
